packages/@lwc/engine-core/src/framework/attributes.ts

```diff
@@ -29,13 +28,43 @@ function offsetPropertyErrorMessage(name: string): string {
 //
 // If you update this list, check for test files that recapitulate the same list. Searching the codebase
 // for e.g. "dropzone" should suffice.
+export const globalHtmlElementPropertyNames = [
+    'accessKey',
+    'accessKeyLabel',
+    'className',
+    'contentEditable',
+    'dataset',
+    'dir',
+    'draggable',
+    'dropzone',
+    'hidden',
+    'id',
+    'inputMode',
+    'isContentEditable',
+    'lang',
+    'offsetHeight',
+    'offsetLeft',
+    'offsetParent',
+    'offsetTop',
+    'offsetWidth',
+    'role',
+    'slot',
+    'spellcheck',
+    'style',
+    'tabIndex',
+    'title',
+    'translate',
+] as const;
+
+// This mapping is deliberately kept separate from the array of property names above because the mapping object is
+// needed only for dev mode, whereas the above array is needed in prod mode too. This makes the bundle size lower.
 export const globalHTMLProperties: {
-    [prop: string]: {
+    [Key in typeof globalHtmlElementPropertyNames[number]]: {
```

Comment:
This TypeScript typing guarantees that the `globalHTMLProperties` object is kept in sync with the array above.

Reply:
I came up with a much simpler solution. This comment is out of date.